src/js/player.js

```diff
@@ -3749,59 +3750,134 @@ class Player extends Component {
-
-    if (isObject(breakpoints)) {
-      assign(this.breakpoints_, breakpoints);
+    if (className) {
+      this.removeClass(className);
     }
+  }
 
-    this.updateCurrentBreakpoint_();
+  /**
+   * Get or set breakpoints on the player.
+   *
+   * Calling this method with an object or `true` will remove any previous
+   * custom breakpoints and start from the defaults again.
+   *
+   * @param  {Object|boolean} [breakpoints]
+   *         If an object is given, it can be used to provide custom
+   *         breakpoints. If `true` is given, will set default breakpoints.
+   *         If this argument is not given, will simply return the current
+   *         breakpoints.
+   *
+   * @param  {number} [breakpoints.tiny]
+   *         The maximum width for the "vjs-layout-tiny" class.
+   *
+   * @param  {number} [breakpoints.xsmall]
+   *         The maximum width for the "vjs-layout-x-small" class.
+   *
+   * @param  {number} [breakpoints.small]
+   *         The maximum width for the "vjs-layout-small" class.
+   *
+   * @param  {number} [breakpoints.medium]
+   *         The maximum width for the "vjs-layout-medium" class.
+   *
+   * @param  {number} [breakpoints.large]
+   *         The maximum width for the "vjs-layout-large" class.
+   *
+   * @param  {number} [breakpoints.xlarge]
+   *         The maximum width for the "vjs-layout-x-large" class.
+   *
+   * @param  {number} [breakpoints.huge]
+   *         The maximum width for the "vjs-layout-huge" class.
+   *
+   * @return {Object}
+   *         An object mapping breakpoint names to maximum width values.
+   */
+  breakpoints(breakpoints) {
+    if (breakpoints !== undefined) {
+      this.breakpoint_ = '';
+      this.breakpoints_ = assign({}, DEFAULT_BREAKPOINTS, breakpoints);
 
-    // If we were not previously supporting breakpoints, add a listener.
-    if (!hadBps) {
-      this.on('playerresize', this.updateCurrentBreakpoint_);
+      // When breakpoint definitions change, we need to update the currently
+      // selected breakpoint.
+      this.updateCurrentBreakpoint_();
     }
+
+    // Clone the breakpoints before returning.
+    return assign(this.breakpoints_);
   }
 
   /**
-   * Get current layout breakpoints for the player.
+   * Get or set a flag indicating whether or not this player should adjust
+   * its UI based on its dimensions.
    *
-   * @return {Object}
-   *         An object describing the current layout breakpoints and their
-   *         max widths.
+   * @param  {boolean} value
+   *         Should be `true` if the player should adjust its UI based on its
+   *         dimensions; otherwise, should be `false`.
+   *
+   * @return {boolean}
+   *         Will be `true` if this player should adjust its UI based on its
+   *         dimensions; otherwise, will be `false`.
    */
-  getBreakpoints() {
-    return this.breakpoints_ && assign(this.breakpoints_);
+  responsive(value) {
+    if (value !== undefined) {
```

Comment:
It would be good to make the getter usage right here with `if (value === undefined)`. That way we get rid of some whitespace.

Reply:
In that case, we also repeat the `return` statement, but you're right - that's closer to our usual pattern.